### ISSUE TITLE
How to set relevant options?

### ISSUE THREAD

**Issue by kero990** [TheAssemblyArmada/Vanilla-Conquer#758]
I compiled the project (RA) on an arm64 linux machine and it worked perfectly, except for the occasional crash.
Now the question is, how do I set up the program? The compiled main program is just an executable file. Running it will directly run redalert at a lower resolution. How can I adjust the resolution? How to try dos mode? etc. I tried modifying redalert.ini, but it doesn't seem to work.
I believe it took so much effort to rewrite this program, shouldn't there be no setting options? There is nothing about it in your documentation. Or if you want to change the settings, you must recompile the program.
Finally, how do I use the units from the expansion, such as field mechanic, shock trooper, in skirmish mode?
In short, I did not find any place to set the main program, nor did I find relevant introduction.
Thank you very much for your project, it is not difficult to play redalert on linux, but it is not easy on arm64.

**Comment by giulianobelinassi** [comment]
Hi, @kero990 

The project is still work in progress, so things are not well documented yet. But I will try to answer your questions

**how do I set up the program?**

Please check the [Installing VanillaRA](https://github.com/TheAssemblyArmada/Vanilla-Conquer/wiki/Installing-VanillaRA) page. It contains information about what and where should you put asset files.

**How can I adjust the resolution?**

Currently, only two resolutions are supported: 640x400 (as the Windows version) and 320x200 (aka DOSMode). Other resolutions are actually one of these two, but scaled. DOSMode still has bugs (mostly in RA). You can enable "DOSMode" by editing ~/.config/vanilla-conquer/vanillara/redalert.ini and setting DOSMode to yes. You will need the lowres1.mix from original discs for it to work. If you want to change the scaled resolution, you can edit redalert.ini as well.

**if you want to change the settings, you must recompile the program?**

No, you don't. Everything can be modified in redalert.ini

**Comment by kero990** [comment]
Oh, thanks so much for your project, I'm finally able to play authentic cnc&ra, not just a modern clone like openra, especially for cross-platform use. Looking forward to the perfect version of that day.
Regarding the resolution, I tried to modify the redalert, but it doesn't seem to work, neither in the main program folder nor in ~/.config/vanilla-conquer/vanillara. If the current resolution is simply scaling, then setting any resolution in full screen should be meaningless.
I read the part about assets. Expansion pieces need to be placed in the relevant folder separately, that is to say, no matter whether the main program contains expansion pieces (such as 3.03), as long as there is no expansion piece folder, there will be no expansion pieces. ? But my main interface has mission buttons for two expansion pieces, do they work?

**Comment by kero990** [comment]
The last question, I see that the project supports the Remastered version, but I also see the use of the Remastered Collection asset file and the -DBUILD_REMASTERRA=ON compile option, what is the relationship between these two methods? Can this compile option compile a binary that completely replaces the main Remastered Collection program to run the Remastered Collection edition directly? Compiling the normal version and just replacing the assets with Remastered, what is the difference in operation? Will there be HD footage? In addition, can the DBUILD_REMASTERRA=ON option be used for linux compilation? I seem to see instructions that only work on win32.

**Comment by giulianobelinassi** [comment]
Remaster builds are just a .dll which you must load in Remaster as a mod. We don't "rebuild" the remaster.

As for resolution: are you running wayland? because it has many problems with resolutions other than the native one.

**Comment by kero990** [comment]
@giulianobelinassi 
Now the problem is, whether I use the Remastered version of the resource, or the original disc resource, I can't load the two expansion packs of redalert. There are buttons for these two expansion packs on the homepage, but they are blank when you click them in, without any missions. In skirmish mode there is also no unit of any of the two expansion packs. I'm sure I configured the directories and files exactly as per the guide at https://github.com/TheAssemblyArmada/Vanilla-Conquer/wiki/Installing-VanillaRA.
And there were no maps even in skirmishes until I copied some files from the CnCRemastered\Data\CNCDATA\RED_ALERT\CUSTOMMAPS\ directory and I didn't know which one did the trick.

So now the project supports extension packs?
In addition, it seems that the release only keeps the latest one, so it is recommended to have an update list, thank you very much for your project again, thank you

**Comment by kero990** [comment]
In addition, since the Remastered version has more language support, has vanillara considered multilingual support? Whether you do it yourself or open the api interface? Even if it's just the ui interface.

**Comment by giulianobelinassi** [comment]
I remember fixing this bug in the Nintendo DS port, but I don't remember porting that to mainline. 

Take a look at expand.cpp in https://github.com/giulianobelinassi/Vanilla-Conquer/commit/0b1ab5162c2de8f8ebf5a7c930febd58352b40fa#diff-065f6d60390d080da0aa92efd5fc9d0c218d561b48cdb63fe3e4121d1350e402L228

**Comment by kero990** [comment]
> I remember fixing this bug in the Nintendo DS port, but I don't remember porting that to mainline.
> 
> Take a look at expand.cpp in [giulianobelinassi@0b1ab51#diff-065f6d60390d080da0aa92efd5fc9d0c218d561b48cdb63fe3e4121d1350e402L228](https://github.com/giulianobelinassi/Vanilla-Conquer/commit/0b1ab5162c2de8f8ebf5a7c930febd58352b40fa#diff-065f6d60390d080da0aa92efd5fc9d0c218d561b48cdb63fe3e4121d1350e402L228)

I replaced the file of the master version with this cpp, but there is still no change when compiling, and the extended task still cannot be loaded.
Waiting for you to fix it to the main line, please @me when you fix it
